src/Protocol/ActivityPub.php

```diff
@@ -277,4 +280,44 @@ public static function isSupportedByContactUrl(string $url, $update = null): boo
 	{
 		return !empty(APContact::getByURL($url, $update));
 	}
+
+	public static function isAcceptedRequester(int $uid = 0): bool
+	{
+		$called_by = System::callstack(1);
+
+		$signer = HTTPSignature::getSigner('', $_SERVER);
+		if (!$signer) {
+			Logger::debug('No signer', ['uid' => $uid, 'agent' => $_SERVER['HTTP_USER_AGENT'] ?? '', 'called_by' => $called_by]);
+			return false;
+		}
+
+		$apcontact = APContact::getByURL($signer);
+		if (empty($apcontact)) {
+			Logger::debug('APContact not found', ['uid' => $uid, 'handle' => $signer, 'called_by' => $called_by]);
+			return false;
+		}
+
+		if (empty($apcontact['gsid'] || empty($apcontact['baseurl']))) {
+			Logger::debug('No server found', ['uid' => $uid, 'signer' => $signer, 'called_by' => $called_by]);
+			return false;
+		}
+
+		// Check added as a precaution. It should not occur.
```

Comment:
Why shouldn't this occur?

Reply:
When we are blocking remote systems, we are doing this on a very low level. Means even a probing for contacts would fail. But a probing is needed for the steps that we perform here in advance. But possibly there are other situations, so this is added here.

Reply:
Since this is a public method, we cannot assume where and when it's going to be used. As a result the check is very much warranted and the I believe the comment is unwarranted.
```suggestion
```

Reply:
Please restore the check, I was only requesting to remove the "It should not occur." comment. 😅

Reply:
Well, I realized that this check is already done in the APContact class.